sormas-ui/src/main/java/de/symeda/sormas/ui/sormastosormas/SormasToSormasListComponent.java

```diff
@@ -171,68 +179,348 @@ private void initLayout(
 	}
 
 	public void reloadList() {
-		sormasToSormasList.reload();
+		UI currentUI = UI.getCurrent();
+		UserDto currentUser = FacadeProvider.getUserFacade().getCurrentUser();
+
+		sormasToSormasList.showPlaceholder(I18nProperties.getString(Strings.sormasToSormasLoadingShares));
+
+		Thread loadSharesThread = new Thread(() -> {
+			try {
+				I18nProperties.setUserLanguage(currentUser.getLanguage());
+				FacadeProvider.getI18nFacade().setUserLanguage(currentUser.getLanguage());
+
+				currentUI.setPollInterval(300);
+
+				List<SormasToSormasShareTree> shareInfos = loadShares.load();
+				String currentServerOrgId = FacadeProvider.getSormasToSormasFacade().getOrganizationId();
+
+				List<SormasToSormasShareInfoDto> shareInfoList = getShareInfoList(shareInfos);
+				SormasToSormasShareInfoDto ownerShare = getOwnerShare(shareInfos);
+				SormasToSormasOriginInfoDto rootOrigin = findRootOriginInfo(shareInfos);
+
+				String ownerOrganizationId = getOwnerOrganizationId(ownerShare, rootOrigin, currentServerOrgId);
+				boolean isOwnedByCurrentOrg = currentServerOrgId.equals(ownerOrganizationId);
+				boolean isOwnedByOrigin = originInfo.getOrganizationId().equals(ownerOrganizationId);
```

Comment:
`originInfo` can be null if the case is not received, which results in a NPE here